### PR TITLE
Real References Addition

### PR DESCRIPTION
**Pull request by Namnodorel**
This PR addresses two things I missed when the earlier PR was merged:
- there was the small but important change of "var -> val", which made compilation fail when in "real references" mode. This is now fixed
- krush does try to go deeper into O2O references where possible, but this only works if the objects on lower levels only have O2O relations, themselves. I added support for including lists in this. Like with the previous PR, this only works when the relation is uni-directional, which is why I implemented it in the "real references" class.

### CODE REVIEW

**Comment by pjagielski** [comment]
Sorry for the delay, I'll review the code later this week

File: example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                    otherColumn = RecipeIngredientsTable.recipeSourceId
+            ).join(
+                    otherTable = IngredientTable,
+                    joinType = JoinType.LEFT,
+                    onColumn = RecipeIngredientsTable.ingredientTargetId,
+                    otherColumn = IngredientTable.id
+            ).selectAll().toMealPlanList().first()
+
+            // then
+            assertThat(selectedMealPlan).isEqualTo(mealPlan)
```
Comment (pjagielski) [inline example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt:51]: I wonder if this works for "copy" mode when you only join Recipe with Ingredients and use `toRecipeList()`?

File: example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        }
+    }
+
+}
```
Comment (pjagielski) [inline example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt:55]: Please ensure you have "line feed at file end on save" enabled ;)

File: example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt (hunk 1 above)
Comment (Namnodorel) [reply example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt:51]: I am not sure whether I understood correctly what you mean - if you only join the recipes with the ingredients, that's a simple list attribute that krush was able to retrieve even before any of my changes.

File: example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt (hunk 2 above)
Comment (Namnodorel) [reply example/src/test/kotlin/pl/touk/krush/realreferences/CookingTest.kt]: Fixed, thanks for the hint ^^

**Pull request merged**